src/nvim/highlight_group.c

```diff
@@ -181,6 +181,7 @@ static const char *highlight_init_both[] = {
   "default link ComplMatchIns    NONE",
   "default link ComplHint        NonText",
   "default link ComplHintMore    MoreMsg",
+  "default link QueryCapture     Title",
```

Comment:
This (and your update to syntax.txt) are for "core" highlights. Not every single plugin needs a new "core" highlight. The highlight group can be defined in `dev.lua`.

Or alternatively, it *would* make sense to define a new core hightlight for a more generic concept. In https://github.com/neovim/neovim/issues/32646 I suggested something like an `Attention` highlight, which fits quite well with the purpose of this query-capture.

Another option is to just use `Visual` instead of `Title`.